concordium-consensus/src/Concordium/ImportExport.hs

```diff
@@ -482,6 +596,14 @@ exportSections dbDir outDir chunkSize genIndex startHeight blockIndex lastWritte
             putStrLn $ "The tree state database does not exist at " ++ treeStateDir
             return (False, Empty)
 
+-- |Action for getting a block (and possibly a finalization index) at a particular height.
+-- Note that for consensus v1 the finalization index is simply @Nothing@ as they do not
+-- exist on this version.
+type GetBlockAt m = BlockHeight -> m (Maybe (BS.ByteString, Maybe FinalizationIndex))
```

Comment:
Could you provide a bit of context as to why we do this slightly hacky version with  an optional FinalizationIndex as opposed to a more "type safe" version. So we can maintain it better in the future.

Reply:
Can you elaborate on when the finalization index will be present? Is it when the block is finalized in <= P5?

Reply:
Elaborated a bit.